components/brave_referrals/browser/brave_referrals_service.cc

```diff
@@ -100,208 +58,30 @@ void BraveReferralsService::Start() {
-        referral_headers_loader_->ResponseInfo()->headers->response_code();
-  if (referral_headers_loader_->NetError() != net::OK || response_code < 200 ||
-      response_code > 299) {
-    const std::string safe_response_body =
-        response_body ? *response_body : std::string();
-    LOG(ERROR) << "Failed to fetch headers from referral server"
-               << ", error: " << referral_headers_loader_->NetError()
-               << ", response code: " << response_code
-               << ", payload: " << safe_response_body
-               << ", url: " << referral_headers_loader_->GetFinalURL().spec();
-    return;
-  }
-
-  base::Optional<base::Value> root =
-      base::JSONReader().ReadToValue(*response_body);
-  if (!root || !root->is_list()) {
-    LOG(ERROR) << "Failed to parse referral headers response";
-    return;
-  }
-  pref_service_->Set(kReferralHeaders, root.value());
-}
-
-void BraveReferralsService::OnReferralInitLoadComplete(
-    std::unique_ptr<std::string> response_body) {
-  int response_code = -1;
-  if (referral_init_loader_->ResponseInfo() &&
-      referral_init_loader_->ResponseInfo()->headers)
-    response_code =
-        referral_init_loader_->ResponseInfo()->headers->response_code();
-  if (referral_init_loader_->NetError() != net::OK || response_code < 200 ||
-      response_code > 299) {
-    const std::string safe_response_body =
-        response_body ? *response_body : std::string();
-    LOG(ERROR) << "Failed to initialize referral"
-               << ", error: " << referral_init_loader_->NetError()
-               << ", response code: " << response_code
-               << ", payload: " << safe_response_body
-               << ", url: " << referral_init_loader_->GetFinalURL().spec();
-    return;
-  }
-
-  base::Optional<base::Value> root =
-      base::JSONReader().ReadToValue(*response_body);
-  if (!root || !root->is_dict()) {
-    LOG(ERROR) << "Failed to parse referral initialization response";
-    return;
-  }
-  if (!root->FindKey("download_id")) {
-    LOG(ERROR)
-        << "Failed to locate download_id in referral initialization response"
-        << ", payload: " << *response_body;
-    return;
-  }
-
-  const base::Value* headers = root->FindKey("headers");
-  if (headers) {
-    pref_service_->Set(kReferralHeaders, *headers);
-  }
-
-  const base::Value* download_id = root->FindKey("download_id");
-  pref_service_->SetString(kReferralDownloadID, download_id->GetString());
-
-  const base::Value* offer_page_url = root->FindKey("offer_page_url");
-  if (offer_page_url) {
-    GURL gurl(offer_page_url->GetString());
-    chrome::ScopedTabbedBrowserDisplayer browser_displayer(
-        ProfileManager::GetLastUsedProfile());
-              content::OpenURLParams open_url_params(gurl, content::Referrer(),
-        WindowOpenDisposition::NEW_FOREGROUND_TAB,
-        ui::PAGE_TRANSITION_AUTO_TOPLEVEL, false);
-    open_url_params.extra_headers = FormatExtraHeaders(headers, gurl);
```

Comment:
Although this function should remain, this line that sets `extra_headers` should go.